src/subtyping.md

```diff
@@ -1,146 +1,143 @@
 
-By default, static types must match *exactly* for a program to compile. As such,
-this code won't compile:
+Let's define that `Sub` is a subtype of `Super` (we'll be using the notation `Sub: Super` throughout this chapter)
 
-<!-- ignore: simplified code -->
-```rust,ignore
-let mr_snuggles: Cat = ...;
-love(mr_snuggles);         // ERROR: expected Animal, found Cat
-```
+What this is suggesting to us is that the set of *requirements* that `Super` defines
+are completely satisfied by `Sub`. `Sub` may then have more requirements.
 
-Mr. Snuggles is a Cat, and Cats aren't *exactly* Animals, so we can't love him! 😿
+An example of simple subtyping that exists in the language are [supertraits](https://doc.rust-lang.org/stable/book/ch19-03-advanced-traits.html?highlight=supertraits#using-supertraits-to-require-one-traits-functionality-within-another-trait)
 
-This is annoying because Cats *are* Animals. They support every operation
-an Animal supports, so intuitively `love` shouldn't care if we pass it a `Cat`.
-We should be able to just **forget** the non-animal parts of our `Cat`, as they
-aren't necessary to love it.
+```rust
+use std::fmt;
 
-This is exactly the problem that *subtyping* is intended to fix. Because Cats are just
-Animals **and more**, we say Cat is a *subtype* of Animal (because Cats are a *subset*
-of all the Animals). Equivalently, we say that Animal is a *supertype* of Cat.
-With subtypes, we can tweak our overly strict static type system
-with a simple rule: anywhere a value of type `T` is expected, we will also
-accept values that are subtypes of `T`.
+pub trait Error: fmt::Display {
+    fn source(&self) -> Option<&(dyn Error + 'static)>;
+    fn description(&self) -> &str;
+    fn cause(&self) -> Option<&dyn Error>;
+}
+```
 
-Or more concretely: anywhere an Animal is expected, a Cat or Dog will also work.
+Here, we have that `Error: fmt::Display` (`Error` is a *subtype* of `Display`),
+because it has all the requirements of `fmt::Display`, plus the `source`/`description`/`cause` functions.
 
-As we will see throughout the rest of this section, subtyping is a lot more complicated
-and subtle than this, but this simple rule is a very good 99% intuition. And unless you
-write unsafe code, the compiler will automatically handle all the corner cases for you.
+However, subtyping in traits is not that interesting.
+Here in the nomicon, we're going to focus more with how subtyping interacts with lifetimes
 
-But this is the Rustonomicon. We're writing unsafe code, so we need to understand how
-this stuff really works, and how we can mess it up.
+Let's define a lifetime to be the simple requirement:
+`'a` defines a region of code in which a value will be alive.
+Now that we have a defined set of requirements for lifetimes, we can define how they relate to each other.
+`'long: 'short` if and only if `'long` defines a region of code that **completely contains** `'short`.
```

Comment:
```suggestion
`'long : 'short` if and only if `'long` defines a region of code that **completely contains** `'short` (_i.e._, `'long ⊇ 'short`)
```

I've added the whitespace to `'a : 'b` kind of bounds, since we're not really within rustfmt-ed code anymore, we're teaching stuff, and `:` for lifetimes represents an (antisymmetrical) binary "boolean" operator, so there is no reason to favor the left hand side over the right hand side.